### DISCUSSION TITLE
Make user download a file from a Buffer

### DISCUSSION THREAD

**Discussion by unkobweb** [adonisjs/core#2875] [Help] [answered]
Hi everyone,

I use Adonis and a S3 API to keep users files, sometimes i want users can encrypt files, the encryption and decryption of the buffer works but now when i download the file from the S3 API, i get the encrypted Buffer, i decrypt it and now how to make this buffer a file to allow the user download it ?

What is the best practice for this case ?
Just use fs to create file, make it download and delete it ?

Best regards

**Comment by thetutlage** [comment, marked as answer]
I don't think you have to create the file at all. Just send the buffer in the response and set the appropriate content-type and other headers ( google will help here )

```ts
response.type('file-content-type-goes-here')
response.send(buffer)
```

**Reply by unkobweb** [reply]
Thanks it works !
